### PR TITLE
RoR Capstone

### PR DESCRIPTION
**Pull request by aizjicod**
## RoR Capstone Project - Me-Money
on this pull request, I will be setting up the project to the main branch
[Project Requirements](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/rails_capstone.md)
[Video](https://www.loom.com/share/7add4aaa57894a7bbb3f47515838fcc9)

## Summary 
- Project Setup
- Create data model and validation according to ERD
- Set up authentication using a devise.
- Create routes, controllers, models, and views with styles
- Add tests

### CODE REVIEW

File: app/views/groups/show.html.erb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        </li>
+      <% end %>
+    <% else %>
+      <li class="group-li empty">
+        haven't made transactions to this category
+      </li>
+    <% end %>
+  </ul>
+  <div>
+    <%= link_to "Add transaction", new_group_expense_path(@group), class: 'add-new-transaction btn' %>
```
Comment (DammyShittu) [inline app/views/groups/show.html.erb]: - [ ] You can personalize the content on the page but you are still expected to follow the design as contained on Behance. Kindly ensure that the new transaction button sticks to the button of the page per the design.

- [ ] I would also recommend you change the navbar heading to "Transactions" when I am on the transactions page to give more context. As you can see I am on the transactions page but the heading is still "Categories".

  Behance design | Your Design
  :---:|:---:
  ioe | uei

File: app/views/expenses/_form.html.erb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    <%= form.number_field :amount, placeholder: 'Amount' %>
+  </div>
+
+  <div class="expense-form-containers checkboxs">
+    <%= form.collection_check_boxes(:group_ids, current_user.groups.order( 'name ASC' ), :id, :name) %>
+  </div>
+  <div class="expense-form-containers submit">
+    <%= form.submit %>
+  </div>
+<% end %>
```
Comment (DammyShittu) [inline app/views/expenses/_form.html.erb:28]: - [ ] When I am on the create new transaction page I click the arrow button on the header I should be taken to the transaction page for the categories. Currently, it navigates to the root page which is incorrect. Kindly fix that.

File: app/views/groups/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+          <% end %>
+        </li>
+      <% end %>
+    <% else %>
+      <li class="group-li empty">
+        no categories yet
+      </li>
+    <% end %>
+  </ul>
+  <%= link_to "New Category", new_group_path, class: 'new-group-link btn' %>
```
Comment (DammyShittu) [inline app/views/groups/index.html.erb]: - [ ] The new category button should stick to the button of the page per he design. Kindly fix that.

  Behance design | Your Design
  :---:|:---:
  ioe | dhue

File: app/views/groups/show.html.erb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  <div id="group-show-information">
+    <p id="show-information-icon"><%= @group.icon %></p>
+    <p id="show-information-name"><%= @group.name %></p>
+    <p id="show-information-total">
+      Total: $<%= get_amount(@group) %>
+    </p>
+  </div>
+  <ul id="group-expenses">
+    <% if @group.expenses.any? %>
+      <% @group.expenses.each do |expense| %>
```
Comment (DammyShittu) [inline app/views/groups/show.html.erb]: - [ ] The expenses should be ordered by the most recently created ones. So we should have something like this.

  ```
  <% @group.expenses.order(created_at: :desc).each do |expense| %>
    ...
  ```

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @aizjicod,

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Mihndim2020 — APPROVED** [review]
# STATUS: APPROVED!!!  🎉 🎉 🎉 :champagne: :champagne: :champagne:
Hi @aizjicod, 

### Good job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file

Your project is complete! Great work on making the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 :champagne: :champagne: :champagne:

Congratulations-Memes-FEATURED

## Optional suggestions

_Check the comments under the review._

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can get to me on slack  using this handle @mihndim.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**